core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -254,6 +254,26 @@ final void contentType(MediaType contentType) {
         set(HttpHeaderNames.CONTENT_TYPE, contentType.toString());
     }
 
+    @Override
+    @Nullable
+    public ContentDisposition contentDisposition() {
+        final String contentDispositionString = get(HttpHeaderNames.CONTENT_DISPOSITION);
+        if (contentDispositionString == null) {
+            return null;
+        }
+
+        try {
+            return ContentDisposition.parse(contentDispositionString);
+        } catch (IllegalArgumentException ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    final void contentDisposition(ContentDisposition contentDisposition) {
+        requireNonNull(contentDisposition, "contentDisposition");
+        set(HttpHeaderNames.CONTENT_DISPOSITION, contentDisposition.toString());
```

Comment:
How about adding `ContentDisposition.asHeaderValue()` and using it instead?

Reply:
You need to update the string conversion logic in `StringMultimap` as well so `asHeaderValue()` is used for string conversion.